File: lecture_practice-nodejs/20221017-getter-setter-2.js
class RgbaColor {
  constructor(red, green, blue, alpha = 1) {
    this.red = red;
    this.green = green;
    this.blue = blue;
    this.alpha = alpha;
  }
  get rgba(){
    return `rgba(${this.red}, ${this.green}, ${this.blue}, ${this.alpha})`;
  }
}

let pracRgbaColor = new RgbaColor(200, 200, 200);
console.log(pracRgbaColor);
console.log(pracRgbaColor.rgba);

File: subject-Python/20221017-getter-setter-2.py
class RgbaColor:
  def __init__(self, red, green, blue, alpha = 1):
    self.red = red
    self.green = green
    self.blue = blue
    self.alpha = alpha
  @property
  def rgba(self):
    return 'rgba({}, {}, {}, {})'.format(self.red, self.green, self.blue, self.alpha)

pracRgbaColor = RgbaColor(200, 200, 200)
print(pracRgbaColor.__dict__)
print(pracRgbaColor.rgba)
